src/iologindata.cpp

```diff
@@ -160,6 +160,35 @@ uint32_t IOLoginData::gameworldAuthentication(const std::string& accountName, co
 	return accountId;
 }
 
+std::list<std::string> IOLoginData::liveCastAuthentication(const std::string& password)
+{
+	std::list<std::string> casts;
+
+	Database* db = Database::getInstance();
+
+	std::ostringstream query;
+	query << "SELECT `player_id`, `name`, `password`, `spectators` FROM `live_casts` WHERE `password` = '" << password << "'";
+	DBResult_ptr result = db->storeQuery(query.str());
+	if (!result) {
+		return casts;
+	}
+
+	query.str(std::string());
+	query << "SELECT `name` FROM `live_casts` WHERE `password` = '" << password << "' ORDER BY `name` ASC";
```

Comment:
Why are you touching the the same database table twice in the same function? TBH I don't see why you would even need to check the database at all to log in a spectator.